### ISSUE TITLE
groupPaths ordering not respected

### ISSUE THREAD

**Issue by andyjessop** [import-js/eslint-plugin-import#2363]
I'm trying to ensure the order of various parent imports, but the order within the parent group seems not to be respected.

```json
"import/order": [1, {
      "groups": ["external", "builtin", "internal", "parent",  "sibling", "index"],
      "newlines-between": "always",
      "pathGroups": [
        {
          "pattern": "~/services/**/*",
          "group": "parent"
        },
        {
          "pattern": "~/components/**/*",
          "group": "parent"
        },
        {
          "pattern": "~hooks/**/*",
          "group": "parent"
        },
      ]
    }],
```

I also tried matching with other patterns such as "~/components", "*components/**/*", etc.

So for instance I'm expecting that this:

```ts
import a from '../../components/x';
import { c } from '../../../hooks/c';
import b from '../../services/b';
```

Should be transformed into this:

```ts
import b from '../../services/b';
import a from '../../components/x';
import { c } from '../../../hooks/c';
```

But I see no errors relating to this.

**Comment by ljharb** [comment]
Indeed, the config you have there is just adding 3 patterns to the parent group - everything within a group is always then sorted by the normal rules.

**Comment by andyjessop** [comment]
Thanks for the quick reply - I misunderstood the docs. So I have a follow up question then, is it possible to do what I want -i.e. to create different groups based on patterns?

**Comment by ljharb** [comment]
The only thing you could really do is a hack - using the supported group names, put each thing in a separate “group”, so that the relative ordering of the “groups” array ordered them.

The point of this rule is to provide a deterministic ordering, grouped by conceptual “kind” of import. Allowing an arbitrary ordering defeats this goal, because a reader of the code would have no idea why the ordering was forced to be idiosyncratic.

**Comment by andyjessop** [comment]
Thanks very much for your responses, you've been really helpful. I'll close this issue.
